### PR TITLE
cephadm: isolate mounts code in methods

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Minor cleanups and a move of inline-logic from the get_container_mounts function for ceph and monitoring daemon types.
This greatly simplifies get_container_mounts and will be used as a basis for future refacotring.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), <s>opened tracker ticket</s> refactoring
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] Relies on existing tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/cephadm/cephadm.py [hunk 1]
```diff
@@ -301,6 +302,52 @@ def get_daemon_args(self) -> List[str]:
...
+            'crash',
+            'ceph-exporter',
+        ]:
+            # these do not search for their keyrings in a data directory
+            mounts[
+                data_dir + '/keyring'
+            ] = '/etc/ceph/ceph.client.%s.%s.keyring' % (
+                ident.daemon_type,
+                ident.daemon_id,
+            )
```
Comment (adk3798) [inline src/cephadm/cephadm.py:341]: is this a black format thing? I find this particular little block a bit harder to read than the original line (even if it was overly long). If it's what the tool does though I can live with it.

**Review by adk3798 — APPROVED** [review]

File: src/cephadm/cephadm.py (hunk 1 above)
Comment (phlogistonjohn) [reply src/cephadm/cephadm.py:341]: Yes. It is a bit unnatural. I think a quick fix would be to move `                data_dir + '/keyring'` to the line above via a variable assignment, and then black ought to make the block less weird looking. I can do that here in this PR or later on in a follow up series. Let me know what you prefer

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-11-08_01:34:19-orch:cephadm-wip-adk-testing-2023-11-07-1631-distro-default-smithi/

8 failures

- 5 of the failures were on tests being added or adjusted by other PRs in the run. Not going to go through each one, but they shouldn't be relevant to the other PRs in the run (test_cephadm_timeout, test_iscsi_container, test_host_drain)
- 1 instance of https://tracker.ceph.com/issues/62958, known issue
- 1 issue in test_cephadm_repos caused by logrotate changes from another PR in the run
- 1 instance of https://tracker.ceph.com/issues/63499, which is new but I've confirmed through an additional run is something wrong on main, not with the PRs in the run.

Shouldn't merge PRs affecting the specific failed tests or things related to them, but a number of other PRs in the run should be okay to merge.

File: src/cephadm/cephadm.py (hunk 1 above)
Comment (adk3798) [reply src/cephadm/cephadm.py:341]: I don't want to hold off merging this over the formatting, so it can be done in a follow up

**Pull request merged**